onnxruntime/core/platform/env.h

```diff
@@ -93,10 +93,16 @@ class Env {
   ///
   /// Caller takes ownership of the result and must delete it eventually
   /// (the deletion will block until fn() stops running).
-  virtual Thread* StartThread(const ThreadOptions& thread_options,
-                              const std::string& name,
+  virtual Thread* StartThread(const ThreadOptions& thread_options, const std::string& name,
                               std::function<void()> fn) const = 0;
 
+  /// file_path must point to a regular file, which can't be a pipe/socket/...
+#ifndef _WIN32
+  virtual common::Status ReadFileAsString(const char* file_path, std::string* out) const = 0;
```

Comment:
>std::string* out [](start = 65, length = 16)

std::string&

Reply:
Let me fix it later. In the next PR, I will remove this parameter.

---
In reply to: [261433704](https://github.com/Microsoft/onnxruntime/pull/520#discussion_r261433704) [](ancestors = 261433704)